magpie/magpie.py

```diff
@@ -65,37 +68,26 @@ def get_request_info(request, default_msg="undefined"):
     return content
 
 
-def init_db():
-    curr_path = os.path.dirname(os.path.abspath(__file__))
-    curr_path = os.path.dirname(curr_path)
-    alembic_ini_path = curr_path+'/alembic.ini'
-    alembic_args = ['-c'+alembic_ini_path, 'upgrade', 'heads']
-    alembic.config.main(argv=alembic_args)
-
-
-def get_database_revision(db_session):
-    s = select(['version_num'], from_obj='alembic_version')
-    result = db_session.execute(s).fetchone()
-    return result['version_num']
-
-
 def main(global_config=None, **settings):
     """
     This function returns a Pyramid WSGI application.
     """
 
-    # Check is database is ready
-    from db import is_database_ready
-    if not is_database_ready():
+    # migrate db as required and check if database is ready
+    try:
+        db.run_database_migration()
+    except ImportError:
+        pass
+    except Exception as e:
+        raise Exception('Database migration failed [{}]'.format(str(e)))
+    if not db.is_database_ready():
```

Comment:
The fct is_database_ready should not be called before doing the migration?

Reply:
`is_database_ready` verifies that all db tables for Group, Users, Services, etc. have been created.
In the event that db migration requires a schema change, a missing table would always fail because migration didn't occur already.

Reply:
Make sense. I was misinterpreting that function as a waiting point for the postgres instance to be ready which obviously is not.